.github/workflows/rust.yml

```diff
@@ -34,7 +34,7 @@ jobs:
 
       - name: cargo fmt
         working-directory: ./rust
-        run: cargo fmt --all -- --check
+        run: cargo +nightly fmt --all -- --check
```

Comment:
Is this required? Because we install the `nightly` toolchain in L12

Reply:
for some reason this is required. i tried it without `+nightly` and the formatting was done like on stable even though the nightly toolchain is installed for this task. i have no idea why, maybe it is some caching issue since the other tasks are installing the stable toolchain.

Reply:
Ah true. By default cargo uses stable. The reason why it worked before is that it was overwritten by then `rust-toolchain` file.